recipes/octomap/byte.patch

```diff
@@ -0,0 +1,37 @@
+diff --git a/octomap/src/binvox2bt.cpp b/octomap/src/binvox2bt.cpp
+index 9b6abec..c2d8048 100644
+--- a/octomap/src/binvox2bt.cpp
++++ b/octomap/src/binvox2bt.cpp
+@@ -56,7 +56,7 @@
+ using namespace std;
+ using namespace octomap;
+ 
+-typedef unsigned char byte;
++typedef unsigned char my_byte;
+ 
+ int main(int argc, char **argv)
+ {
+@@ -227,8 +227,8 @@ int main(int argc, char **argv)
+         cout.flush();
+             
+         // read voxel data
+-        byte value;
+-        byte count;
++        my_byte value;
```

Comment:
Trying to use the C++17 `byte` causes a compilation error below on the `*input >> value`:

https://github.com/OctoMap/octomap/blob/bf3b3949a99c13ef34d11dc8bc23dcd6680a9f3c/octomap/src/binvox2bt.cpp#L230-L238

<details>
<summary>Just in case the 300+ lines compilation error</summary>
```
/tmp/octomap/octomap/src/binvox2bt.cpp: In function 'int main(int, char**)':
/tmp/octomap/octomap/src/binvox2bt.cpp:236:16: error: no match for 'operator>>' (operand types are 'std::ifstream {aka std::basic_ifstream<char>}' and 'std::byte')
         *input >> value;    // read the linefeed char
         ~~~~~~~^~~~~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:120:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__istream_type& (*)(std::basic_istream<_CharT, _Traits>::__istream_type&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(__istream_type& (*__pf)(__istream_type&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:120:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__istream_type& (*)(std::basic_istream<char>::__istream_type&) {aka std::basic_istream<char>& (*)(std::basic_istream<char>&)}'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:124:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__ios_type& (*)(std::basic_istream<_CharT, _Traits>::__ios_type&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>; std::basic_istream<_CharT, _Traits>::__ios_type = std::basic_ios<char>]
       operator>>(__ios_type& (*__pf)(__ios_type&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:124:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__ios_type& (*)(std::basic_istream<char>::__ios_type&) {aka std::basic_ios<char>& (*)(std::basic_ios<char>&)}'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:131:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::ios_base& (*)(std::ios_base&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(ios_base& (*__pf)(ios_base&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:131:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::ios_base& (*)(std::ios_base&)'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:168:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(bool&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(bool& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:168:7: note:   no known conversion for argument 1 from 'std::byte' to 'bool&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:172:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(short int&) [with _CharT = char; _Traits = std::char_traits<char>]
       operator>>(short& __n);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:172:7: note:   no known conversion for argument 1 from 'std::byte' to 'short int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:175:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(short unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned short& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:175:7: note:   no known conversion for argument 1 from 'std::byte' to 'short unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:179:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(int&) [with _CharT = char; _Traits = std::char_traits<char>]
       operator>>(int& __n);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:179:7: note:   no known conversion for argument 1 from 'std::byte' to 'int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:182:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned int& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:182:7: note:   no known conversion for argument 1 from 'std::byte' to 'unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:186:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:186:7: note:   no known conversion for argument 1 from 'std::byte' to 'long int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:190:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:190:7: note:   no known conversion for argument 1 from 'std::byte' to 'long unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:195:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long long int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:195:7: note:   no known conversion for argument 1 from 'std::byte' to 'long long int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:199:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long long unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned long long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:199:7: note:   no known conversion for argument 1 from 'std::byte' to 'long long unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:214:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(float&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(float& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:214:7: note:   no known conversion for argument 1 from 'std::byte' to 'float&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:218:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(double&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(double& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:218:7: note:   no known conversion for argument 1 from 'std::byte' to 'double&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:222:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long double&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long double& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:222:7: note:   no known conversion for argument 1 from 'std::byte' to 'long double&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:235:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(void*&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(void*& __p)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:235:7: note:   no known conversion for argument 1 from 'std::byte' to 'void*&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:259:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__streambuf_type*) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__streambuf_type = std::basic_streambuf<char>]
       operator>>(__streambuf_type* __sb);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:259:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__streambuf_type* {aka std::basic_streambuf<char>*}'
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/string:53:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:49:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/basic_string.tcc:1465:5: note: candidate: template<class _CharT, class _Traits, class _Alloc> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&)
     operator>>(basic_istream<_CharT, _Traits>& __in,
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/basic_string.tcc:1465:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   mismatched types 'std::__cxx11::basic_string<_CharT, _Traits, _Alloc>' and 'std::byte'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:991:0,
                 from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:963:5: note: candidate: template<class _CharT2, class _Traits2> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, _CharT2*)
     operator>>(basic_istream<_CharT, _Traits>& __in, _CharT* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:963:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   mismatched types '_CharT2*' and 'std::byte'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:991:0,
                 from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:931:5: note: candidate: template<class _CharT, class _Traits> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, _CharT&)
     operator>>(basic_istream<_CharT, _Traits>& __in, _CharT& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:931:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::byte')
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:756:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, unsigned char&)
     operator>>(basic_istream<char, _Traits>& __in, unsigned char& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:756:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   cannot convert 'value' (type 'std::byte') to type 'unsigned char&'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:761:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, signed char&)
     operator>>(basic_istream<char, _Traits>& __in, signed char& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:761:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   cannot convert 'value' (type 'std::byte') to type 'signed char&'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:803:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, unsigned char*)
     operator>>(basic_istream<char, _Traits>& __in, unsigned char* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:803:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   cannot convert 'value' (type 'std::byte') to type 'unsigned char*'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:808:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, signed char*)
     operator>>(basic_istream<char, _Traits>& __in, signed char* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:808:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   cannot convert 'value' (type 'std::byte') to type 'signed char*'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: note: candidate: template<class _Istream, class _Tp> typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_istream<_Istream>, std::__is_extractable<typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type, _Tp&&, void> >::value, typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type>::type std::operator>>(_Istream&&, _Tp&&)
     operator>>(_Istream&& __is, _Tp&& __x)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: note:   template argument deduction/substitution failed:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream: In substitution of 'template<class _Istream, class _Tp> typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_istream<_Istream>, std::__is_extractable<typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type, _Tp&&, void> >::value, typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type>::type std::operator>>(_Istream&&, _Tp&&) [with _Istream = std::basic_ifstream<char>&; _Tp = std::byte&]':
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19:   required from here
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: error: no type named 'type' in 'struct std::enable_if<false, std::basic_istream<char>&>'
In file included from /tmp/octomap/octomap/include/octomap/OcTreeBaseImpl.h:42:0,
                 from /tmp/octomap/octomap/include/octomap/OccupancyOcTreeBase.h:44,
                 from /tmp/octomap/octomap/include/octomap/OcTree.h:38,
                 from /tmp/octomap/octomap/include/octomap/octomap.h:37,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:52:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bitset:1462:5: note: candidate: template<class _CharT, class _Traits, long unsigned int _Nb> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, std::bitset<_Nb>&)
     operator>>(std::basic_istream<_CharT, _Traits>& __is, bitset<_Nb>& __x)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bitset:1462:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:19: note:   mismatched types 'std::bitset<_Nb>' and 'std::byte'
         *input >> value;    // read the linefeed char
                   ^~~~~
In file included from /tmp/octomap/octomap/include/octomap/AbstractOcTree.h:37:0,
                 from /tmp/octomap/octomap/include/octomap/AbstractOccupancyOcTree.h:37,
                 from /tmp/octomap/octomap/include/octomap/OccupancyOcTreeBase.h:45,
                 from /tmp/octomap/octomap/include/octomap/OcTree.h:38,
                 from /tmp/octomap/octomap/include/octomap/octomap.h:37,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:52:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/cstddef:134:5: note: candidate: template<class _IntegerType> constexpr std::__byte_op_t<_IntegerType> std::operator>>(std::byte, _IntegerType)
     operator>>(byte __b, _IntegerType __shift) noexcept
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/cstddef:134:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:236:9: note:   cannot convert '* input' (type 'std::ifstream {aka std::basic_ifstream<char>}') to type 'std::byte'
         *input >> value;    // read the linefeed char
         ^~~~~~
/tmp/octomap/octomap/src/binvox2bt.cpp:239:20: error: no match for 'operator>>' (operand types are 'std::ifstream {aka std::basic_ifstream<char>}' and 'std::byte')
             *input >> value >> count;
             ~~~~~~~^~~~~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:120:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__istream_type& (*)(std::basic_istream<_CharT, _Traits>::__istream_type&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(__istream_type& (*__pf)(__istream_type&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:120:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__istream_type& (*)(std::basic_istream<char>::__istream_type&) {aka std::basic_istream<char>& (*)(std::basic_istream<char>&)}'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:124:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__ios_type& (*)(std::basic_istream<_CharT, _Traits>::__ios_type&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>; std::basic_istream<_CharT, _Traits>::__ios_type = std::basic_ios<char>]
       operator>>(__ios_type& (*__pf)(__ios_type&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:124:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__ios_type& (*)(std::basic_istream<char>::__ios_type&) {aka std::basic_ios<char>& (*)(std::basic_ios<char>&)}'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:131:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(std::ios_base& (*)(std::ios_base&)) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(ios_base& (*__pf)(ios_base&))
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:131:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::ios_base& (*)(std::ios_base&)'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:168:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(bool&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(bool& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:168:7: note:   no known conversion for argument 1 from 'std::byte' to 'bool&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:172:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(short int&) [with _CharT = char; _Traits = std::char_traits<char>]
       operator>>(short& __n);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:172:7: note:   no known conversion for argument 1 from 'std::byte' to 'short int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:175:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(short unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned short& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:175:7: note:   no known conversion for argument 1 from 'std::byte' to 'short unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:179:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(int&) [with _CharT = char; _Traits = std::char_traits<char>]
       operator>>(int& __n);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:179:7: note:   no known conversion for argument 1 from 'std::byte' to 'int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:182:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned int& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:182:7: note:   no known conversion for argument 1 from 'std::byte' to 'unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:186:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:186:7: note:   no known conversion for argument 1 from 'std::byte' to 'long int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:190:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:190:7: note:   no known conversion for argument 1 from 'std::byte' to 'long unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:195:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long long int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:195:7: note:   no known conversion for argument 1 from 'std::byte' to 'long long int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:199:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long long unsigned int&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(unsigned long long& __n)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:199:7: note:   no known conversion for argument 1 from 'std::byte' to 'long long unsigned int&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:214:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(float&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(float& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:214:7: note:   no known conversion for argument 1 from 'std::byte' to 'float&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:218:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(double&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(double& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:218:7: note:   no known conversion for argument 1 from 'std::byte' to 'double&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:222:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(long double&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(long double& __f)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:222:7: note:   no known conversion for argument 1 from 'std::byte' to 'long double&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:235:7: note: candidate: std::basic_istream<_CharT, _Traits>::__istream_type& std::basic_istream<_CharT, _Traits>::operator>>(void*&) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__istream_type = std::basic_istream<char>]
       operator>>(void*& __p)
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:235:7: note:   no known conversion for argument 1 from 'std::byte' to 'void*&'
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:259:7: note: candidate: std::basic_istream<_CharT, _Traits>& std::basic_istream<_CharT, _Traits>::operator>>(std::basic_istream<_CharT, _Traits>::__streambuf_type*) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_istream<_CharT, _Traits>::__streambuf_type = std::basic_streambuf<char>]
       operator>>(__streambuf_type* __sb);
       ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:259:7: note:   no known conversion for argument 1 from 'std::byte' to 'std::basic_istream<char>::__streambuf_type* {aka std::basic_streambuf<char>*}'
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/string:53:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:49:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/basic_string.tcc:1465:5: note: candidate: template<class _CharT, class _Traits, class _Alloc> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&)
     operator>>(basic_istream<_CharT, _Traits>& __in,
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/basic_string.tcc:1465:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   mismatched types 'std::__cxx11::basic_string<_CharT, _Traits, _Alloc>' and 'std::byte'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:991:0,
                 from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:963:5: note: candidate: template<class _CharT2, class _Traits2> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, _CharT2*)
     operator>>(basic_istream<_CharT, _Traits>& __in, _CharT* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:963:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   mismatched types '_CharT2*' and 'std::byte'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:991:0,
                 from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:931:5: note: candidate: template<class _CharT, class _Traits> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, _CharT&)
     operator>>(basic_istream<_CharT, _Traits>& __in, _CharT& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bits/istream.tcc:931:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::byte')
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:756:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, unsigned char&)
     operator>>(basic_istream<char, _Traits>& __in, unsigned char& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:756:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   cannot convert 'value' (type 'std::byte') to type 'unsigned char&'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:761:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, signed char&)
     operator>>(basic_istream<char, _Traits>& __in, signed char& __c)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:761:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   cannot convert 'value' (type 'std::byte') to type 'signed char&'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:803:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, unsigned char*)
     operator>>(basic_istream<char, _Traits>& __in, unsigned char* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:803:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   cannot convert 'value' (type 'std::byte') to type 'unsigned char*'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:808:5: note: candidate: template<class _Traits> std::basic_istream<char, _Traits>& std::operator>>(std::basic_istream<char, _Traits>&, signed char*)
     operator>>(basic_istream<char, _Traits>& __in, signed char* __s)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:808:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   cannot convert 'value' (type 'std::byte') to type 'signed char*'
             *input >> value >> count;
                       ^~~~~
In file included from /home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/fstream:38:0,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:50:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: note: candidate: template<class _Istream, class _Tp> typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_istream<_Istream>, std::__is_extractable<typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type, _Tp&&, void> >::value, typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type>::type std::operator>>(_Istream&&, _Tp&&)
     operator>>(_Istream&& __is, _Tp&& __x)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: note:   template argument deduction/substitution failed:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream: In substitution of 'template<class _Istream, class _Tp> typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_istream<_Istream>, std::__is_extractable<typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type, _Tp&&, void> >::value, typename std::__is_convertible_to_basic_istream<_Tp>::__istream_type>::type std::operator>>(_Istream&&, _Tp&&) [with _Istream = std::basic_ifstream<char>&; _Tp = std::byte&]':
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23:   required from here
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/istream:980:5: error: no type named 'type' in 'struct std::enable_if<false, std::basic_istream<char>&>'
In file included from /tmp/octomap/octomap/include/octomap/OcTreeBaseImpl.h:42:0,
                 from /tmp/octomap/octomap/include/octomap/OccupancyOcTreeBase.h:44,
                 from /tmp/octomap/octomap/include/octomap/OcTree.h:38,
                 from /tmp/octomap/octomap/include/octomap/octomap.h:37,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:52:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bitset:1462:5: note: candidate: template<class _CharT, class _Traits, long unsigned int _Nb> std::basic_istream<_CharT, _Traits>& std::operator>>(std::basic_istream<_CharT, _Traits>&, std::bitset<_Nb>&)
     operator>>(std::basic_istream<_CharT, _Traits>& __is, bitset<_Nb>& __x)
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/bitset:1462:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:23: note:   mismatched types 'std::bitset<_Nb>' and 'std::byte'
             *input >> value >> count;
                       ^~~~~
In file included from /tmp/octomap/octomap/include/octomap/AbstractOcTree.h:37:0,
                 from /tmp/octomap/octomap/include/octomap/AbstractOccupancyOcTree.h:37,
                 from /tmp/octomap/octomap/include/octomap/OccupancyOcTreeBase.h:45,
                 from /tmp/octomap/octomap/include/octomap/OcTree.h:38,
                 from /tmp/octomap/octomap/include/octomap/octomap.h:37,
                 from /tmp/octomap/octomap/src/binvox2bt.cpp:52:
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/cstddef:134:5: note: candidate: template<class _IntegerType> constexpr std::__byte_op_t<_IntegerType> std::operator>>(std::byte, _IntegerType)
     operator>>(byte __b, _IntegerType __shift) noexcept
     ^~~~~~~~
/home/local/lesteve/miniconda3/envs/hpp-fcl/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/cstddef:134:5: note:   template argument deduction/substitution failed:
/tmp/octomap/octomap/src/binvox2bt.cpp:239:13: note:   cannot convert '* input' (type 'std::ifstream {aka std::basic_ifstream<char>}') to type 'std::byte'
             *input >> value >> count;
             ^~~~~~
/tmp/octomap/octomap/src/binvox2bt.cpp:242:35: error: no match for 'operator+' (operand types are 'int' and 'std::byte')
                 end_index = index + count;
                             ~~~~~~^~~~~~~
/tmp/octomap/octomap/src/binvox2bt.cpp:271:45: error: no match for 'operator==' (operand types are 'std::byte' and 'int')
                       if(mark_free || value == 1) {
                                       ~~~~~~^~~~
/tmp/octomap/octomap/src/binvox2bt.cpp:272:60: error: no match for 'operator==' (operand types are 'std::byte' and 'int')
                           tree->updateNode(endpoint, value == 1, true);
                                                      ~~~~~~^~~~
/tmp/octomap/octomap/src/binvox2bt.cpp:279:26: error: could not convert 'value' from 'std::byte' to 'bool'
                 if (value) nr_voxels += count;
                          ^
/tmp/octomap/octomap/src/binvox2bt.cpp:279:38: error: no match for 'operator+=' (operand types are 'unsigned int' and 'std::byte')
                 if (value) nr_voxels += count;
                            ~~~~~~~~~~^~~~~~~~

```

</details>